### PR TITLE
OIDC Logout Bug

### PR DESCRIPTION
**Pull request by zofskeez**
It was discovered that when logging out when previously authenticated via OIDC or JWT auth methods the `with` query param is set to `token` which hydrates the auth form with the `token` auth method selected rather than the previously logged in method. To ensure that the selected auth method persists I updated the auth service to use `sessionStorage` for this piece of state.

### CODE REVIEW

File: ui/app/components/auth-form.js [hunk 1]
```diff
@@ -206,12 +211,15 @@ export default Component.extend(DEFAULTS, {
   showLoading: or('isLoading', 'authenticate.isRunning', 'fetchMethods.isRunning', 'unwrapToken.isRunning'),
 
   authenticate: task(
-    waitFor(function* (backendType, data) {
-      let clusterId = this.cluster.id;
+    waitFor(function* (backend, data) {
```
Comment (hashishaw) [inline ui/app/components/auth-form.js:214]: I do like that the previous name was clear that this is the backend type, not the backend name. Could we add a comment or keep the old name here?

**Review by hashishaw — APPROVED** [review]
LGTM! Thanks 🚀

File: ui/app/components/auth-form.js [hunk 2]
```diff
@@ -118,14 +130,7 @@ export default Component.extend(DEFAULTS, {
     'selectedAuth',
     'selectedAuthIsPath',
     function () {
```
[the remark this answers is not in the document]
Comment (zofskeez) [reply ui/app/components/auth-form.js:132]: I thought so too and tried it but `this` becomes undefined.

File: ui/app/components/auth-form.js (hunk 1 above)
Comment (zofskeez) [reply ui/app/components/auth-form.js:214]: Sure I can change it back. The auth service `authenticate` method has it named `backend` so I thought I would align it that way.

**Pull request merged**